tests/app/main/test_briefs.py

```diff
@@ -2086,3 +2092,203 @@ def test_evaluation_methods_shown_with_a_or_an(self, data_api_client):
         doc = html.fromstring(res.get_data(as_text=True))
         assert len(doc.xpath('//li[contains(normalize-space(text()), "a work history")]')) == 1
         assert len(doc.xpath('//li[contains(normalize-space(text()), "an interview")]')) == 1
+
+
+@mock.patch("app.main.views.briefs.data_api_client")
+class TestResponseResultPage(ResponseResultPageBothFlows):
+
+    def setup_method(self, method):
+        super(TestResponseResultPage, self).setup_method(method)
+        self.brief['briefs']['niceToHaveRequirements'] = []
+        self.brief['briefs']['dayRate'] = []
+        self.brief_responses = {
+            'briefResponses': [
+                {
+                    'essentialRequirementsMet': True,
+                    'essentialRequirements': [
+                        {'evidence': 'Did a thing'},
+                        {'evidence': 'Produced a thing'},
+                        {'evidence': 'Did a thing'}
+                    ]
+                }
+            ]
+        }
+
+    def test_view_response_result_page_escapes_essential_and_nice_to_have_requirements(self, data_api_client):
+        self.brief['briefs']['essentialRequirements'] = [
+            '<h1>Essential one with xss</h1>',
+            '**Essential two with markdown**'
+        ]
+        self.brief['briefs']['niceToHaveRequirements'] = [
+            '<h1>n2h one with xss</h1>',
+            '**n2h two with markdown**'
+        ]
+        self.brief_responses['briefResponses'][0]['niceToHaveRequirements'] = [
+            {
+                'yesNo': True,
+                'evidence': 'Did a thing'
+            },
+            {
+                'yesNo': False
+            }
+        ]
+
+        data_api_client.get_brief.return_value = self.brief
+        data_api_client.get_framework.return_value = self.framework
+        data_api_client.is_supplier_eligible_for_brief.return_value = True
+        data_api_client.find_brief_responses.return_value = self.brief_responses
+
+        res = self.client.get('/suppliers/opportunities/1234/responses/result')
+        data = res.get_data(as_text=True)
+        doc = html.fromstring(data)
+
+        assert "&lt;h1&gt;Essential one with xss&lt;/h1&gt;" in data
+        assert "&lt;h1&gt;n2h one with xss&lt;/h1&gt;" in data
+        assert len(doc.xpath('//h1')) == 1
+
+        assert "**Essential two with markdown**" in data
+        assert "<strong>Essential two with markdown</strong>" not in data
+        assert "**n2h two with markdown**" in data
+        assert "<strong>n2h two with markdown</strong>" not in data
+
+    def test_view_response_result_submitted_ok(self, data_api_client):
```

Comment:
Makes sense for both old and new flows to be variants of the same test. I'll make the changes.